### ISSUE TITLE
800-108 KDF: counter after iterator

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#408]
The old CAVS tool defined a "counter after iterator" option for the feedback mode. The ACVP SPEC does not seem to allow that.

May I ask how to define the option of "counter after iterator"?

**Comment by celic** [comment]
That would correspond to `"after fixed data"`.

**Comment by smuellerDD** [comment]
Am Montag, 25. Juni 2018, 17:44:53 CEST schrieb Chris Celi:

Hi Chris,

> That would correspond to `"after fixed data"`.

That is what I thought so too. But my feedback mode KDF does not validate with 
ACVP even though it successfully validates with the NIST CAVS trial vectors 
with "counter after iterator".

Ciao
Stephan

**Comment by smuellerDD** [comment]
Just tested it again, all my FB results are rejected.

Please note that the old CAVS tool has "after fixed data" and "after iterator" whereas the ACVP server only has "after fixed data":

```
[PRF=CMAC_AES128]
[CTRLOCATION=AFTER_FIXED]
[RLEN=32_BITS]
```

and

```
[PRF=CMAC_AES192]
[CTRLOCATION=AFTER_ITER]
[RLEN=8_BITS]
```

**Comment by celic** [comment]
After Iter must be `"middle fixed data"` then. Sorry, it's been a bit since I've looked at that part of KDF.

**Comment by smuellerDD** [comment]
Am Montag, 25. Juni 2018, 19:04:22 CEST schrieb Chris Celi:

Hi Chris,

> After Iter must be `"middle fixed data"` then. Sorry, it's been a bit since
> I've looked at that part of KDF.

SPEC: "middle fixed data is not valid for feedback nor double pipeline 
iterator".

Ciao
Stephan

**Comment by celic** [comment]
I'll take a deeper look at this shortly. :^)

**Comment by smuellerDD** [comment]
Thanks, that fixed it.

I will keep it open to prop up the spec.

**Comment by smuellerDD** [comment]
Fixed with pull request #417.
